### DISCUSSION TITLE
Type input without using zod/validation?

### DISCUSSION THREAD

**Discussion by brhx** [trpc/trpc#978] [Q&A] [answered]
Hello 👋 

Is there a way to type the input object without using validation? I'm trying to re-create the behaviour of this Blitz example: 

https://github.com/blitz-js/blitz/blob/63af540c879fe71dfad561d814b4da57dcafe576/examples/auth/app/projects/queries/getProjects.ts#L9

**Comment by KATT** [comment, marked as answer]
Yes, but you should really, really **not** do this

```tsx
function passThroughAsNumber(input: unknown) {
  return input as number;
}
const router = trpc.router().query('num', {
  input: passThroughAsNumber,
  resolve({ input }) {
    return {
      input,
    };
  },
});
```

**Reply by brhx** [reply]
That's totally fair! Though by the time they're authenticated for this router the input is pretty strongly trusted. As I'm using zod everywhere else I also found I can use the following:

```ts
const router = trpc.router().query("getProjects", {
  input: z.custom<GetProjectsInput>(),
  resolve({ input }) {
    return { input };
  },
});
```
